hpecp/role.py

```diff
@@ -0,0 +1,158 @@
+# (C) Copyright [2020] Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from __future__ import absolute_import
+from .logger import Logger
+from .exceptions import (
+    ContainerPlatformClientException,
+    APIException,
+    APIItemNotFoundException,
+    APIItemConflictException,
+)
+
+import json
+from operator import attrgetter
+
+# from tabulate import tabulate
+# from enum import Enum
+# import polling
+import re
+import six
+import sys
+
+try:
+    basestring
+except NameError:
+    basestring = str
+
+
+class RoleController:
+    """This is the main class that users will interact with to work with roles.
+
+    An instance of this class is available in the client.ContainerPlatformClient with the attribute name
+    :py:attr:`role <.client.ContainerPlatformClient.role>`.  The methods of this class can be
+    invoked using `client.role.method()`.  See the example below:
+
+    Example::
+
+        client = ContainerPlatformClient(...).create_session()
+        client.role.get()
+
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def get(self, role_id):
```

Comment:
Can you please add unit tests for these methods?

Reply:
Will do this.
Thanks for the guidance.

Reply:
Sure thing Chris! Thank you :")